src/tests/account/ethereum/test_eth_account.cairo

```diff
@@ -426,34 +424,37 @@ fn test_public_key_setter_different_account() {
 fn test_public_key_setter_and_getter_camel() {
     let mut state = COMPONENT_STATE();
     let public_key = ETH_PUBKEY();
-    let new_public_key = NEW_ETH_PUBKEY();
-
-    testing::set_contract_address(ACCOUNT_ADDRESS());
-    testing::set_caller_address(ACCOUNT_ADDRESS());
+    let key_pair = KEY_PAIR();
+    let contract_address = test_address();
 
+    start_cheat_caller_address(contract_address, contract_address);
     state.initializer(public_key);
-    utils::drop_event(ACCOUNT_ADDRESS());
 
     let current = state.getPublicKey();
     assert_eq!(current, public_key);
 
-    state.setPublicKey(new_public_key, get_accept_ownership_signature());
+    let mut spy = spy_events();
+
+    let signature = get_accept_ownership_signature(contract_address, ETH_PUBKEY(), key_pair);
+    state.setPublicKey(key_pair.public_key, signature);
 
-    assert_event_owner_removed(ACCOUNT_ADDRESS(), public_key);
-    assert_only_event_owner_added(ACCOUNT_ADDRESS(), new_public_key);
+    spy.assert_event_owner_removed(contract_address, public_key);
+    spy.assert_only_event_owner_added(contract_address, key_pair.public_key);
 
     let public_key = state.getPublicKey();
-    assert_eq!(public_key, new_public_key);
+    assert_eq!(public_key, key_pair.public_key);
 }
 
 #[test]
 #[should_panic(expected: ('EthAccount: unauthorized',))]
 fn test_public_key_setter_different_account_camel() {
     let mut state = COMPONENT_STATE();
-    testing::set_contract_address(ACCOUNT_ADDRESS());
-    testing::set_caller_address(CALLER());
+    let key_pair = KEY_PAIR();
 
-    state.setPublicKey(NEW_ETH_PUBKEY(), get_accept_ownership_signature());
+    start_cheat_caller_address(test_address(), CALLER());
+
+    let signature = get_accept_ownership_signature(test_address(), ETH_PUBKEY(), key_pair);
```

Comment:
A local variable `contract_address` would be helpful since `test_address()` is called twice